sopel/coretasks.py

```diff
@@ -432,20 +485,32 @@ def _periodic_send_who(bot):
 @sopel.module.thread(False)
 @sopel.module.unblockable
 def track_join(bot, trigger):
-    if trigger.nick == bot.nick and trigger.sender not in bot.channels:
-        bot.write(('TOPIC', trigger.sender))
+    channel = trigger.sender
 
-        bot.privileges[trigger.sender] = dict()
-        bot.channels[trigger.sender] = Channel(trigger.sender)
-        _send_who(bot, trigger.sender)
+    # is it a new channel?
+    if channel not in bot.channels:
+        LOGGER.info('Channel joined: %s', channel)
+        bot.privileges[channel] = dict()
+        bot.channels[channel] = Channel(channel)
+
+    # did we just joined?
```

Comment:
```suggestion
    # did *we* just join?
```

Though I'm not sure how a joined channel could be `not in bot.channels` with a user who is not Sopel. If Sopel has joined, it *must* be a new channel, unless we're leaving remnants of privilege tracking around after Sopel PARTs/QUITs.

Reply:
I had some issue with the MODE event being processed first on some very rare case - maybe it was when I played with the `thread=True` and nothing else - and I don't want to assume that the event sequence will always be the same.

In that particular case, I feel like it's better to be less strict: in the end, a WHO request is sent, privilege are reset to 0, and if the server starts acting weird, at least it won't break Sopel with something that we can handle fine.

Reply:
Fair. I'll leave this unresolved until you've applied (or discarded) the suggestion.